server/src/main/java/org/opensearch/action/support/clustermanager/TransportClusterManagerNodeAction.java

```diff
@@ -217,6 +218,15 @@ public boolean shouldRetry(Exception e) {
             return false;
         }
 
+        /**
+         * If tasks gets timed out in retrying on throttling,
+         * it should send cluster event timeout exception.
+         */
+        @Override
+        public Exception getTimeoutException(Exception e) {
+            return new ProcessClusterEventTimeoutException(request.masterNodeTimeout, actionName);
+        }
+
```

Comment:
We are pretty much discarding the underlying exception which is a serious impairment to debuggability. Can we wrap exception causes instead

Reply:
For debuggability, `RetrayableAction` will add the underlying exception in suppressed exception list and which will be logged on data node. So in data node logs, throttling exception will be logged. Master will also logs on throttling exceptions.

We just want to keep user experience consistent, so to user it will be thrown as `ProcessClusterEventTimeoutException`.

Reply:
Maybe add another overloaded constructor for and keep the underlying cause as is.
```
 public ProcessClusterEventTimeoutException(TimeValue timeValue, String source, Exception ex) {
        super("failed to process cluster event (" + source + ") within " + timeValue, ex);
    }
```

Reply:
Actually when tasks is getting timed out due to throttling, those throttling exception is returned as suppressed exception in process cluster event timeout exception. `RetryableAction` does this for us, it stores upto last 4 exceptions and add it in suppressed exception. 

From that user will know that task is getting timed out due to throttling till we expose the throttling exception to user.

Do we still need to add underlying cause in event timeout exception?

sample exception which is returned to user:

```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "process_cluster_event_timeout_exception",
        "reason" : "failed to process cluster event (indices:admin/mapping/put) within 30s",
        "suppressed" : [
          {
            "type" : "cluster_manager_throttling_exception",
            "reason" : "Throttling Exception : Limit exceeded for put-mapping"
          },
          {
            "type" : "cluster_manager_throttling_exception",
            "reason" : "Throttling Exception : Limit exceeded for put-mapping"
          },
          {
            "type" : "cluster_manager_throttling_exception",
            "reason" : "Throttling Exception : Limit exceeded for put-mapping"
          }
        ]
      }
    ],
    "type" : "process_cluster_event_timeout_exception",
    "reason" : "failed to process cluster event (indices:admin/mapping/put) within 30s",
    "suppressed" : [
      {
        "type" : "cluster_manager_throttling_exception",
        "reason" : "Throttling Exception : Limit exceeded for put-mapping"
      },
      {
        "type" : "cluster_manager_throttling_exception",
        "reason" : "Throttling Exception : Limit exceeded for put-mapping"
      },
      {
        "type" : "cluster_manager_throttling_exception",
        "reason" : "Throttling Exception : Limit exceeded for put-mapping"
      }
    ]
  },
  "status" : 503
}
```